### DISCUSSION TITLE
What all should go into the next debian packet?

### DISCUSSION THREAD

**Discussion by ftasnetamot** [yrutschle/sslh#469] [Ideas]
Hello,

this discussion should collect things, which should go to the next debian packet.  My idea is, that sslh users contribute with things from their daily experience. Hopefully we can motivate the debian maintainer, to give some wishes, what he likes to see, what could make his life easier (and hopefully will bring some more updates).
Furthermore sslh users could contribute the one or other pull request, as from my current perspective documentation is currently a bigger point. This looks like something, more of us could do together.

From my perspective, the current sys-v init-script needs a good rework:
- with a restart it kills all current sessions (at least for the sslh-fork implementation), but this is not really needed.
  deleting two lines will make your life happier.
- it has a reload-function implemented, which sends a HUP-signal to sslh. But sslh is not aware of this signal, so only the main process gets killed, while the first daughter process stays alive.  This makes a next restart impossible, as the main listening address is used by this process, but that does not belong to the store PID 

And, opening this discussion here, one thing, which is a must in any distribution repository:
Give hints, how and where users can contribute to the source of the project.
- sslh should give a good hint, when called with an invalid parameter.  BTW there is right now no --help, -h -? or any other parameter, associated with help. Fortunately every wrong parameter gives the help information ;-)
- the man page should give this information

**Comment by onehalf3570** [comment]
reload option is, in my opinion, is extremely valuable - right now I get
```
# systemctl reload sslh
Failed to reload sslh.service: Job type reload is not applicable for unit sslh.service
```

So I have to do a restart if/when I change the configuration - but this closes the established connections ;( - at least for the sslh-ev flavour, which is expected (sslh-fork works, but it won't work with UDP connections)

**Comment by yrutschle** [comment]
I remember looking into dynamic reload of configuration, and concluding it was a major work, when this can be achieved fairly simply by mixing e.g. sslh-ev with fork for specific protocols. Typically I use sslh-ev, but set up forking for ssh, as there aren' t that many ssh connections it doesn' t matter in terms of performance, but then your ssh sessions survive restarting sslh.

**Reply by onehalf3570** [reply]
Thanks! Missed that config param.
